pkg/core/config/tenant.go

```diff
@@ -10,14 +10,15 @@ import (
 // TenantConfiguration is a mock struct of tenant configuration
 //go:generate msgp -tests=false
 type TenantConfiguration struct {
-	DBConnectionStr string                   `msg:"DATABASE_URL" envconfig:"DATABASE_URL" json:"DATABASE_URL"`
-	APIKey          string                   `msg:"API_KEY" envconfig:"API_KEY" json:"API_KEY"`
-	MasterKey       string                   `msg:"MASTER_KEY" envconfig:"MASTER_KEY" json:"MASTER_KEY"`
-	AppName         string                   `msg:"APP_NAME" envconfig:"APP_NAME" json:"APP_NAME"`
-	CORSHost        string                   `msg:"CORS_HOST" envconfig:"CORS_HOST" default:"*"`
-	TokenStore      TokenStoreConfiguration  `json:"TOKEN_STORE" msg:"TOKEN_STORE"`
-	UserProfile     UserProfileConfiguration `json:"USER_PROFILE" msg:"USER_PROFILE"`
-	UserAudit       UserAuditConfiguration   `json:"USER_AUDIT" msg:"USER_AUDIT"`
+	DBConnectionStr           string                   `msg:"DATABASE_URL" envconfig:"DATABASE_URL" json:"DATABASE_URL"`
+	APIKey                    string                   `msg:"API_KEY" envconfig:"API_KEY" json:"API_KEY"`
+	MasterKey                 string                   `msg:"MASTER_KEY" envconfig:"MASTER_KEY" json:"MASTER_KEY"`
+	AppName                   string                   `msg:"APP_NAME" envconfig:"APP_NAME" json:"APP_NAME"`
+	CORSHost                  string                   `msg:"CORS_HOST" envconfig:"CORS_HOST" default:"*"`
+	TokenStore                TokenStoreConfiguration  `json:"TOKEN_STORE" msg:"TOKEN_STORE"`
+	UserProfile               UserProfileConfiguration `json:"USER_PROFILE" msg:"USER_PROFILE"`
+	UserAudit                 UserAuditConfiguration   `json:"USER_AUDIT" msg:"USER_AUDIT"`
+	UserProfileRecordStoreURL string                   `msg:"USER_PROFILE_RECORD_STORE_URL" envconfig:"USER_PROFILE_RECORD_STORE_URL" json:"USER_PROFILE_RECORD_STORE_URL"`
```

Comment:
This config should be under `UserProfileConfiguration`.

For config that have more than one implementation, I think you can take skygear v1 asset store as reference.

https://github.com/SkygearIO/skygear-server/blob/master/pkg/server/skyconfig/config.go#L152
https://github.com/SkygearIO/skygear-server/blob/master/pkg/server/skyconfig/config.go#L441

---

`UserProfileSkygearURL` or `UserProfileRecordStoreURL`? @carmenlau @rickmak any comment?

Reply:
@Steven-Chan I updated it to 

```
type UserProfileConfiguration struct {
	ImplName     string `msg:"IMPLEMENTATION" envconfig:"USER_PROFILE_IMPL_NAME" json:"IMPLEMENTATION"`
	ImplStoreURL string `msg:"IMPL_STORE_URL" envconfig:"USER_PROFILE_IMPL_STORE_URL" json:"IMPL_STORE_URL"`
}
```

let me know if it is a proper name. Thanks.